src/diffpy/labpdfproc/tests/test_functions.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import pytest
+
+from diffpy.labpdfproc.functions import Gridded_circle
+
+params1 = [
+    [[1, 3, 1, {(0.0, -1.0), (0.0, 0.0), (-1.0, 0.0), (1.0, 0.0), (0.0, 1.0)}, 5]],
+    [[0.2, 8, 1, {(-0.08571428571428572, -0.14285714285714285), (-0.02857142857142858, -0.14285714285714285), (0.02857142857142858, -0.14285714285714285), (0.08571428571428574, -0.14285714285714285), (-0.14285714285714285, -0.08571428571428572), (-0.08571428571428572, -0.08571428571428572), (-0.02857142857142858, -0.08571428571428572), (0.02857142857142858, -0.08571428571428572), (0.08571428571428574, -0.08571428571428572), (0.14285714285714285, -0.08571428571428572), (-0.14285714285714285, -0.02857142857142858), (-0.08571428571428572, -0.02857142857142858), (-0.02857142857142858, -0.02857142857142858), (0.02857142857142858, -0.02857142857142858), (0.08571428571428574, -0.02857142857142858), (0.14285714285714285, -0.02857142857142858), (-0.14285714285714285, 0.02857142857142858), (-0.08571428571428572, 0.02857142857142858), (-0.02857142857142858, 0.02857142857142858), (0.02857142857142858, 0.02857142857142858), (0.08571428571428574, 0.02857142857142858), (0.14285714285714285, 0.02857142857142858), (-0.14285714285714285, 0.08571428571428574), (-0.08571428571428572, 0.08571428571428574), (-0.02857142857142858, 0.08571428571428574), (0.02857142857142858, 0.08571428571428574), (0.08571428571428574, 0.08571428571428574), (0.14285714285714285, 0.08571428571428574), (-0.08571428571428572, 0.14285714285714285), (-0.02857142857142858, 0.14285714285714285), (0.02857142857142858, 0.14285714285714285), (0.08571428571428574, 0.14285714285714285)}, 32]]
+]
+
+@pytest.mark.parametrize('params1', params1)
+def test_get_grid_points(params1):
+    for param_set in params1:
+        radius, n_points_on_diameter, mu, expected_grid, expected_total_points = param_set
+        actual_gs = Gridded_circle(radius=radius, n_points_on_diameter=n_points_on_diameter, mu=mu)
+        assert actual_gs.grid == expected_grid
+        assert actual_gs.total_points_in_grid == expected_total_points
+
+
+
+# values of distances are correct but im a bit confused about the order of distances
+params2 = [
+    [[1, 3, 1, [25, 90, 140],
```

Comment:
you need to run `black` on everything to get the layout right.  But when you install `pre-commit` it will do it for you when you commit.  You will have to run the commit multiple times as when it runs black the commit fails, so just rerun it (if flake8 fails you have to fix the errors manually then rerun the commit).